contracts/FungibleTokenSwitchboard.cdc

```diff
@@ -69,6 +69,31 @@ pub contract FungibleTokenSwitchboard {
             emit VaultCapabilityAdded(type: vaultRef.getType())
         }
 
+        /// addNewVaultsByPath adds a number of new fungible token receiver 
+        ///                    capabilities by using the paths where they are
+        ///                    stored
+        ///
+        /// Parameters: paths: The paths where the public capabilities are stored
+        ///             address: The address of the owner of the capabilities
+        ///
+        pub fun addNewVaultsByPath(paths: [PublicPath], address: Address) {
+            // Get the account where the public capabilities are stored
+            let owner = getAccount(address)
+            // For each path, get the saved capability and store it 
+            // into the switchboard's receiver capabilities dictionary 
+            for path in paths {
+                let capability = owner.getCapability<&{FungibleToken.Receiver}>(path)
+                // Borrow a reference to the vault pointed by the capability we want
+                // to store inside the switchboard
+                let vaultRef = capability.borrow() 
+                    ?? panic ("Cannot borrow reference to vault from capability")
```

Comment:
Can we maybe make this panic free ? I had in my mind to keep somewhere list of common paths, and initialize with that.

Reply:
I'm not 100% sure on what will be preferable for users, since if we make it panic free, users will need to check after which vaults where actually added. @joshuahannan what do you think about it?

Reply:
I mean with event I guess it is more manageable, otherwise event is useless as is, we need to remove that. But it would be nice to have a whitelist of paths ( for example at marketplace ) 

also I strongly favor not overwriting the capability. I mean if I already forwarding flow, I should remove to redirect it to another capability.

Reply:
You are absolutely right, using the event consumers could know which of their vaults were added to the switchboard. 

About overwriting that's something worth a little discussion, but probably not being able to overwrite it by mistake makes a lot of sense.